### PR TITLE
Validation

### PR DESCRIPTION
**Pull request by thedac**
README update.
Small bug fixes.
Initial functional testing infrastructure.

### CODE REVIEW

File: src/tox.ini [hunk 1]
```diff
@@ -1,53 +1,77 @@
-# Source charm: ./src/tox.ini
+# Classic charm: ./tox.ini
 # This file is managed centrally by release-tools and should not be modified
 # within individual charm repos.
 [tox]
-envlist = pep8
+envlist = pep8,py27,py35,py36
 skipsdist = True
+skip_missing_interpreters = True
```
Comment (ryan-beisner) [inline src/tox.ini]: :-(

**Review by ryan-beisner — COMMENTED** [review]
**Re: tox ini files**
If you do an env list of pep8, py3,
remove the bit about skipping missing interpreters,
and add a py3 section that declares the interpreter as python3,
and leave the py35 and py36 sections in place,
and remove all py27 sections and references,
and kill any call to /bin/true with fire,
the gates will work, plus the developer story on any-py3-system should just work,
 and I will be quite happy.

File: src/tox.ini (hunk 1 above)
Comment (ryan-beisner) [reply src/tox.ini]: +1 from me on all of the rest.  Please go ahead and land it when the tox ini items are addressed.  Thanks!  :cat2:

File: src/README.md [hunk 2]
```diff
@@ -180,15 +245,23 @@ In order to take the above into account several objects need to be created:
...
+    # arbitrary.
+    openstack identity provider create --remote-id https://samltest.id/saml/idp samltest
+    # Use the rules.json created above.
+    openstack mapping create --rules rules.json samltest_mapping
+    # The name should be mapped or saml here and must match the configuration
+    # setting protocol-name. We recommend using "mapped"
+    openstack federation protocol create mapped --mapping samltest_mapping --identity-provider samltest
     # list related projects
     openstack federation project list
+    # Note and auto generated domain has been created. This is where auto
```
Comment (pengale) [inline src/README.md:258]: Type: "Note and auto" should be "Note an auto"

**Review by pengale — COMMENTED** [review]
Caught one typo. Changes otherwise make sense and look good.

**Pull request merged**